test/utils/pods/pods.go

```diff
@@ -3,146 +3,19 @@ package pods
-			[]string{"/bin/bash", "-c", fmt.Sprintf(hugePageCommand, writeSize)}, []string{},
-		),
-		corev1.RestartPolicyNever,
-	)
-
-	pod.Spec.NodeSelector = map[string]string{
-		"kubernetes.io/hostname": nodeName,
-	}
-
-	// Resource request
-	pod.Spec.Containers[0].Resources.Requests = corev1.ResourceList{}
-	pod.Spec.Containers[0].Resources.Requests["memory"] = resource.MustParse("1Gi")
-	pod.Spec.Containers[0].Resources.Requests["hugepages-1Gi"] = resource.MustParse("4Gi")
-	pod.Spec.Containers[0].Resources.Requests["cpu"] = *resource.NewQuantity(int64(4), resource.DecimalSI)
-
-	// Resource limit
-	pod.Spec.Containers[0].Resources.Limits = corev1.ResourceList{}
-	pod.Spec.Containers[0].Resources.Limits["memory"] = resource.MustParse("1Gi")
-	pod.Spec.Containers[0].Resources.Limits["hugepages-1Gi"] = resource.MustParse("4Gi")
-	pod.Spec.Containers[0].Resources.Limits["cpu"] = *resource.NewQuantity(int64(4), resource.DecimalSI)
-
-	// Hugepages volume mount
-	pod.Spec.Containers[0].VolumeMounts = []corev1.VolumeMount{{Name: "hugepages", MountPath: "/dev/hugepages "}}
-
-	// Security context capabilities
-	pod.Spec.Containers[0].SecurityContext = &corev1.SecurityContext{RunAsUser: pointer.Int64Ptr(0),
-		Capabilities: &corev1.Capabilities{Add: []corev1.Capability{"IPC_LOCK"}}}
-
-	// Hugepages volume
-	pod.Spec.Volumes = []corev1.Volume{{Name: "hugepages",
-		VolumeSource: corev1.VolumeSource{
-			EmptyDir: &corev1.EmptyDirVolumeSource{
-				Medium: corev1.StorageMediumHugePages}}}}
-
-	return pod
-}
 
-// WaitForDeletion waits until the pod will be removed from the cluster
-func WaitForDeletion(cs *testclient.ClientSet, pod *corev1.Pod, timeout time.Duration) error {
-	return wait.PollImmediate(time.Second, timeout, func() (bool, error) {
-		_, err := cs.Pods(pod.Namespace).Get(context.Background(), pod.Name, metav1.GetOptions{})
-		if errors.IsNotFound(err) {
-			return true, nil
-		}
-		return false, nil
-	})
-}
-
-// WaitForCondition waits until the pod will have specified condition type with the expected status
-func WaitForCondition(cs *testclient.ClientSet, pod *corev1.Pod, conditionType corev1.PodConditionType, conditionStatus corev1.ConditionStatus, timeout time.Duration) error {
-	return wait.PollImmediate(time.Second, timeout, func() (bool, error) {
-		updatePod, err := cs.Pods(pod.Namespace).Get(context.Background(), pod.Name, metav1.GetOptions{})
-		if err != nil {
-			return false, nil
-		}
-
-		for _, c := range updatePod.Status.Conditions {
-			if c.Type == conditionType && c.Status == conditionStatus {
-				return true, nil
-			}
-		}
-		return false, nil
-	})
-}
+	testclient "github.com/openshift/ptp-operator/test/utils/client"
+)
 
 // GetLog connects to a pod and fetches log
 func GetLog(p *corev1.Pod) (string, error) {
-	req := testclient.Client.Pods(p.Namespace).GetLogs(p.Name, &corev1.PodLogOptions{})
+	req := testclient.Client.Pods(p.Namespace).GetLogs(p.Name, &corev1.PodLogOptions{Container: "linuxptp-daemon-container"})
```

Comment:
Can you make this generic and pass the container name as a parameter? Otherwise it's' GetLogFromDaemonContainer